### ISSUE TITLE
Experimental Design [New tabs design]

### ISSUE THREAD

**Issue by skylerschain** [department-of-veterans-affairs/vets-design-system-documentation#2346]
## What
> This an updated version of tabs which we have [deployed](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68964) on the Claim Status Tool. We worked on this new design originally because of an [a11y issue with the previous tab design](https://github.com/department-of-veterans-affairs/va.gov-team/issues/41449). This tab style is more accessible and fits better with the visual style of the latest VADS components. <br>
**Previous**
<img width="431" alt="Screenshot 2023-12-18 at 12 25 52 PM" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/95102058/3969cc75-e2d2-4e00-bc1b-be2ffd1c57a1"> <br>
**New**
<img width="625" alt="Screenshot 2023-12-18 at 12 26 03 PM" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/95102058/a2baed75-ed10-425c-bbc3-6c2cc068fe61"> <br>

## Purpose
> Tabs allow content-rich pages to be subdivided into more manageable sections, creating a simpler and more organized user experience.

## Usage
> Tabs can be used for pages or products that are especially robust in terms of feature set or content volume.

## Behavior
> Clicking or tapping a tab takes the user to a dedicated page. Each tab is a separate react-router `Link`. Navigating between tabs take the user to a distinct URL.

## Examples
>  <img width="651" alt="Screenshot 2023-12-18 at 3 04 43 PM" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/95102058/92365971-064f-42fa-bd47-2042be4dbed6"> 

## Accessibility
> This redesign was meant to resolve an [existing 508 defect](https://github.com/department-of-veterans-affairs/va.gov-team/issues/41449) with the previous version of tabs. The new design was a collaboration between @skylerschain @jerekshoe and a11y experts Josh Kim and @davidakennedy. 

> NOTE: There was feedback earlier about how dark the gray line in the tab component is. We've updated it to a darker shade of gray and are now using $color-vads-color-base-lighter (#dfe1e2). 

## Guidance
> Tabs are not necessary for most pages. First try keeping content on a single page with headers, but if you have simply too much content and clearly different _types_ of content, then tabs are a great option. <br>
Specs: <br>
[Tabs in our spec](https://www.figma.com/file/F8U4wddaFouUPVd4mGBMDI/CST---Evidence-Submission---Spec?type=design&node-id=80-1335&mode=design&t=ZACg4wBz9GNx8hja-4) (mobile and desktop) <br> 
[Hover state behavior for tabs](https://www.figma.com/file/F8U4wddaFouUPVd4mGBMDI/CST---Evidence-Submission---Spec?type=design&node-id=80-1336&mode=design&t=ZACg4wBz9GNx8hja-4)

## Research (optional)
> These tabs tested well in our Evidence Submission usability testing earlier this year. 
[Research Findings](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/claim-appeal-status/research/2023-05-Evidence-Submission/research-findings.md)
[Prototype](https://www.sketch.com/s/98e35645-34eb-40cf-80c1-6c4952943584/prototype/252DFC7E-C58A-478F-A521-5C082886DA75/a/252DFC7E-C58A-478F-A521-5C082886DA75)

## Code (optional)
> Include any existing code.
```jsx
<nav aria-label="Claim">
  <ul className="tabs">
    <li>
      <IndexLink
        aria-current="true"
        activeClassName="tab--current"
        className="tab"
        to={`your-claims/${id}/status`}
      >
        <span>Status</span>
      </IndexLink>
    </li>
    <li>
      <IndexLink
        aria-current="false"
        activeClassName="tab--current"
        className="tab"
        to={`your-claims/${id}/files`}
      >
        <span>Files</span>
      </IndexLink>
    </li>
    <li>
      <IndexLink
        aria-current="false
        activeClassName="tab--current"
        className="tab"
        to={`your-claims/${id}/details`}
      >
        <span>Details</span>
      </IndexLink>
    </li>
  </ul>
<nav>
```

## Next steps
You may present your work to the Design System Council at an upcoming meeting. If you do not or cannot attend the Design Council Meeting, you can opt to get an asynchronous approval. 

Submit requests to join an upcoming Design System Council meeting in #platform-design-system.  

During the meeting, the Design System Council Working Group will evaluate the request and make a decision. 

If your request is approved, you can [add your component or pattern to the system](https://design.va.gov/about/contributing-to-the-design-system/add-a-component-or-pattern-once-approved). If you have any questions on how to add your component or pattern to the system, please reach out to the Design System Team at #platform-design-system.

**Comment by caw310** [comment]
@skylerschain can you attend the DSC meeting on Friday, Jan 5 to talk about this?

**Comment by skylerschain** [comment]
@caw310 What time are those again?

**Comment by skylerschain** [comment]
@caw310 Are there specific questions I can prepare to answer, or is it more of a general conversation?

**Comment by caw310** [comment]
@skylerschain , there isn't anything specific. Usually, it is just the request presenting the problem and how this design solves that problem and how it would be useful for other teams.

**Comment by humancompanion-usds** [comment]
In DSC on 1/5/24 we agreed that if Skyler and DK came up with some parameters around the behavior of the links within the tabs at 400% zoom then I could switch our existing guidance to point to this new version. The parameters are:

- How long (in CSS ex would be ideal but px are find too) the tab name can be
- The minimum and maximum number of tabs (leaning towards minimum of 3 and max of however many will fit at 400% zoom)
- Directions on how the tab name should relate or be connected to the header for the tab (i.e. the current tab section header at the top of the content for that tab) for a11y 

We also discussed that there are the following now in existence on va.gov:

### In-page navigation

- Tabs
- On this page

**Behavior - Navigation**

- Same page, refreshed data/content (Appointments future state using parameters)
- Distinct URL and page, but displayed within the context of the original page (Claim status tool, existing Appointments)

### Navigation

- Link
- Main Menu
- Sidenav

**Behavior - Navigation**

- New URL/page, page displays as different page (MHV secondary nav)

The [Appointments tabs were requested already](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1699). We'll want to consolidate on one Tab design. I prefer the design in this request so leaning towards that.

**Comment by humancompanion-usds** [comment]
One thing I want to be clear on, normally the Design System Team would take on this work of taking the component, standardizing, making it bullet proof for uses in other contexts, etc. But given that we are in the midst of a large and all consuming upgrade to USWDS v3 we asked Skyler and DK to do some of this work for us. Once we're past this upgrade the DST will take on this type of work again.

**Comment by skylerschain** [comment]
One update: I did recently tweak [the component itself in Figma](https://www.figma.com/file/lbRZz68qGRGTdUigcoJq3a/Local-BMT-Components?type=design&node-id=33-29&mode=design&t=ih16UEupWZaCIeqh-4) to make it more responsive to different tab labels. 

We haven't figured out the answers to the questions posed above yet. But we're just getting through a major FE work phase so I think we can reprioritize this soon!

**Comment by davidakennedy** [comment]
@skylerschain and I have completed some more work on this. It's in a place where we think it's ready to have the design system team take a look, ask questions, and/or move the work forward.

Here's the [design file with some of the specs](https://www.figma.com/design/lbRZz68qGRGTdUigcoJq3a/Local-Benefits-Tools-Components?node-id=365-143&t=PyXN2L49SKmmHDTL-0).

And here some overall thoughts on the options we explored in the design file:

## Overall accessibility for options

Of the three options, Option 1 is the best choice from an accessibility perspective.

**Why**: The menu has a simple design on mobile screens, and doesn't hide menu items behind a toggle. Past research with other similar menus on VA.gov performed well.

## Other overall experience-related thoughts

- How consistent are the icons used across the site for controls that are menus/expandable? For example, the main navigation uses the hamburger icon, but the logged-in menu uses the chevron.
- A label for any collapsable menu will be a challenge, and should be standardized so a consistent approach gets implemented if a menu is used in multiple places on the site.
- All of the mobile versions in each option require more research. Currently, the Benefits Management Tools team doesn't have a need for more than three menu items in the Claims Status Tool, and hasn't researched beyond that number of menu items.

### Option 1: Stacked on Mobile

As mentioned, this option is the best from an accessibility perspective, and presents a good starting point for a new component.

#### Pros

- Mobile design is simple, and doesn't hide menu items behind a toggle.
- A similar menu performed well in testing on the Digital Health Modernization team when it tested a secondary navigation. [Findings include](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/secondary-nav/Comparison-study/2024-02-topline-summary.md): "Participants struggled, for a variety of reasons, with the collapsed navigation"
  - Some participants did not seem to notice the menu
  - Some participants did not come back to the menu after engaging with it
  - A couple of the participants stated that persistent navigation/icon design (Prototype B) was “much improved” and “significantly easier.”

#### Cons

- The use of truncation on smaller screens will be a challenge. It's generally difficult to make work for screen readers, and challenging for those with cognitive disabilities.
- The component would take up more vertical space on mobile/small screens.

### Option 2: Horizontal Scrolling on Mobile

This version isn't viable because the horizontal scrolling on mobile would violate [Success Criterion 1.4.10](https://www.w3.org/TR/WCAG22/#reflow).

#### Pros

- Doesn't take up any more vertical space on small screens.

#### Cons

- Violates [Success Criterion 1.4.10](https://www.w3.org/TR/WCAG22/#reflow).
- On mobile screens, the scrollbar may not show up unless the container is touched first. This might mean Veterans may not know there are more menu items.

### Option 3

This version shows as much of the menu as possible, and only hides a portion of items on smaller screens. It doesn't avoid hiding items though.

#### Pros

- Still shows many of the menu items.

#### Cons

- Hides some menu items, which haven't performed well in [past research](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/secondary-nav/Comparison-study/2024-02-topline-summary.md).

---

Let me know if you all have any questions on any of this.

**Comment by thejordanwood** [comment]
The Sitewide team has also been working on a new tab design that follows the [USWDS segmented button group](https://designsystem.digital.gov/button-group/button-group--segmented/). @LWWright7 Has requested that I add this here because it relates to this experimental design ticket. 

## What
The Sitewide team is looking to make updates to the [Facility Locator](https://www.va.gov/find-locations). On mobile, there are currently tabs to switch between the list view and the map view. We would like to update the tabs to the [USWDS segmented button group](https://designsystem.digital.gov/button-group/button-group--segmented/).

**Previous**
<img width="157" alt="Image" src="https://github.com/user-attachments/assets/c3f86ac2-eebf-4523-91bc-7624460fe332" />

**New**
<img width="206" alt="Image" src="https://github.com/user-attachments/assets/8d3b161c-9cf1-4e44-99f8-decb835d7938" />

## Purpose & Usage
This follows the same purpose and usage outlined in the ticket. 

## Behavior
Segmented buttons can be used as a switch between different views. This behavior will be on mobile only and will alternate between the list view and map view 

## Accessibility
Accessibility guidance is outlined on the [USWDS segmented button group page](https://designsystem.digital.gov/button-group/button-group--segmented/). 

## Specs

[Figma file for mobile map designs](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=0-4407&t=PbzANzxHjIYSKA4p-1)

## Research (optional)
We plan on testing several different updates we're making to the Facility Locator, including the segmented button group. 

cc: @Agile6MSkinner @laflannery

**Comment by caw310** [comment]
Notes from the DSC meeting: 
- Use mobile styling and test it. 
- Text has to be bold on selected item. 
- VADS doesn't have a component for this. You will have to build as a team and contribute back. 

[Meeting notes](https://docs.google.com/document/d/15Y-p-dTYbvkoZeoyYoYsX4fT86D_8vYUoRyWVlZ9sts/edit?usp=sharing)

**Comment by thejordanwood** [comment]
@Agile6MSkinner @davidmpickett @eselkin This shows the decisions that were made about the segmented button group in the DSC meeting today. Tagging so that you can follow along with this ticket if needed.

**Comment by JoelCalumpong** [comment]
Hi, just noting that SM on VA.gov has been using the hanging links / new tabs proposed in this ticket since the secondary nav was implemented in the health apt.

![Image](https://github.com/user-attachments/assets/321ce838-c2c0-48c6-9206-559cf7f378ff)

SM's looks a little different (HR is a different length) - SM needs to adjust to match.

**Comment by jilladams** [comment]
Sitewide did get the Segmented button group built and it could be considered to pull into VADS. 
Visible on https://staging.va.gov/find-locations if you are at a mobile breakpoint. (it's behind a feature toggle that is disabled on Prod til after we research: `facilityLocatorMobileMapUpdate`)

PR for the code is here: https://github.com/department-of-veterans-affairs/vets-website/pull/34468
